src/references/game_events.md

```diff
@@ -168,6 +170,231 @@ The `eventID` value can be retrieved by copying the **Game Event Id** for the ev
+local RSVP_TRIGGER = script:GetCustomProperty("Trigger"):WaitForObject()
+local RSVP_UI = script:GetCustomProperty("UIContainer"):WaitForObject()
+local RSVP_BUTTON = script:GetCustomProperty("UIEventRSVPButton"):WaitForObject()
+local EXIT_UI_BUTTON = script:GetCustomProperty("ExitButton"):WaitForObject()
+local EVENT_ID = script:GetCustomProperty("EventId")
+
+--Set the RSVP Button to the correct Game Event ID
+RSVP_BUTTON.eventId = EVENT_ID
+
+local localPlayer = Game.GetLocalPlayer()
+local inTrigger = false
+
+local function CloseUI()
+    RSVP_UI.visibility = Visibility.FORCE_OFF
+
+    if inTrigger then
+        RSVP_TRIGGER.isInteractable = true
+    else
+        RSVP_TRIGGER.isInteractable = false
+    end
+
+    UI.SetCursorVisible(false)
+    UI.SetCanCursorInteractWithUI(false)
+end
+
+local function OnInteracted(trigger, obj)
+    if inTrigger and Object.IsValid(obj) and obj:IsA("Player") and obj == localPlayer then
+        RSVP_UI.visibility = Visibility.FORCE_ON
+        RSVP_TRIGGER.isInteractable = false
+
+        UI.SetCursorVisible(true)
+        UI.SetCanCursorInteractWithUI(true)
+    end
+end
+
+local function OnExitTrigger(trigger, obj)
+    if Object.IsValid(obj) and obj:IsA("Player") and obj == localPlayer then
+        inTrigger = false
+        CloseUI()
+    end
+end
+
+local function OnEnterTrigger(trigger, obj)
+    if Object.IsValid(obj) and obj:IsA("Player") and obj == localPlayer then
+        RSVP_TRIGGER.isInteractable = true
+        inTrigger = true
+    end
+end
+
+EXIT_UI_BUTTON.clickedEvent:Connect(CloseUI)
+
+RSVP_TRIGGER.interactedEvent:Connect(OnInteracted)
+RSVP_TRIGGER.endOverlapEvent:Connect(OnExitTrigger)
+RSVP_TRIGGER.beginOverlapEvent:Connect(OnEnterTrigger)
+```
+
+<div class="mt-video" style="width:100%">
+    <video autoplay muted playsinline controls loop class="center" style="width:100%">
+        <source src="/img/GameEvents/rsvp_example.mp4" type="video/mp4" />
+    </video>
+</div>
+
+### Event Participants
+
+Creators may want a list of player names that participated in an event. One way to do this is store each player in a [Leaderboard](../api/leaderboards.md) if the event is currently active. Here's an example that uses [Chat](../api/chat.md) commands to display a **Leaderboard** of players that were in the game while an event was occurring.
+
+```lua
+local LEADERBOARD = script:GetCustomProperty("Leaderboard")
+local EVENT_ID = script:GetCustomProperty("EventID")
+
+local eventData = CorePlatform.GetGameEvent(EVENT_ID)
+local eventStartTime = 0
+local eventEndTime = 0
+
+--Only the creator has the ability to print event participants
+local creatorName = "DoubleABattery"
+
+local checkEventTask = nil
+
+--Access the start and end time for the event
+if eventData ~= nil then
+    eventStartTime = eventData:GetStartDateTime().secondsSinceEpoch
+    eventEndTime = eventData:GetEndDateTime().secondsSinceEpoch
+end
+
+function CheckEvent()
```

Comment:
For consistency across docs, change all functions to be local since they do not need to be accessed outside of the context.